homeassistant/components/light/lifx.py

```diff
@@ -140,31 +113,35 @@ def convert_rgb_to_hsv(rgb):
 class LIFXLight(Light):
     """Representation of a LIFX light."""
 
-    def __init__(self, liffy, ipaddr, name, power, hue, saturation, brightness,
-                 kelvin):
+    def __init__(self, device):
         """Initialize the light."""
-        _LOGGER.debug("LIFXLight: %s %s", ipaddr, name)
-
-        self._liffylights = liffy
-        self._ip = ipaddr
-        self.set_name(name)
-        self.set_power(power)
-        self.set_color(hue, saturation, brightness, kelvin)
+        self.device = device
+        self.updated_event = asyncio.Event()
+        self.blocker = None
+        self.postponed_update = None
+        self._available = True
+        self.set_power(device.power_level)
+        self.set_color(*device.color)
 
     @property
-    def should_poll(self):
-        """No polling needed for LIFX light."""
-        return False
+    def available(self):
```

Comment:
Instead of being all this fancy, you can also just have an `available` property on the entity (defined in `__init__`). No need abstracting it in a setter.